browser/net/brave_request_handler.cc

```diff
@@ -87,6 +87,9 @@ void BraveRequestHandler::SetupCallbacks() {
 
 #if BUILDFLAG(ENABLE_IPFS)
   if (base::FeatureList::IsEnabled(ipfs::features::kIpfsFeature)) {
+    callback = base::BindRepeating(
+        ipfs::OnBeforeURLRequest_LocalIPFSNodeSchemeManager);
+    before_url_request_callbacks_.push_back(callback);
```

Comment:
We don't need two before_url_request_callbacks, you can split them up internally in [ipfs_redirect_network_delegate_helper.cc](https://github.com/brave/brave-core/pull/14883/files#diff-5a208de77796edde3a97ada956a4cd34b8b55426786d2920c14931553653ed2d)

Reply:
They have different intents: existing one translates ipfs://(ipns://) urls to http://<gateway base url>/ipfs(ipns).
This one serves correct http scheme for localhost node.
Wouldn't it be better to separate actions to different callbacks?

Reply:
i guess there is not much difference from the technical perspective, but ya i'd also have larger entrypoints here and split functions closer to the actual work